nxc/protocols/nfs.py

```diff
@@ -273,19 +273,38 @@ def get_file(self):
             self.auth["uid"] = attrs["attributes"]["uid"]
             dir_handle = mnt_info["mountinfo"]["fhandle"]
 
-            # Get the file handle and read the file data
+            # Get the file handle
             dir_data = self.nfs3.lookup(dir_handle, file_name, auth=self.auth)
             file_handle = dir_data["resok"]["object"]["data"]
-            file_data = self.nfs3.read(file_handle, auth=self.auth)
 
-            if "resfail" in file_data:
-                raise Exception("Insufficient Permissions")
-            else:
-                data = file_data["resok"]["data"]
+            # Get the file attributes (size)
+            file_attrs = self.nfs3.getattr(file_handle, auth=self.auth)
+            file_size = file_attrs["attributes"]["size"]
+
+            # Handle files over 32768 bytes
+            offset = 0
+            chunk_size = 32768  # 32KB
+            eof = False
```

Comment:
In the read response is also the variable `eof` present. Can you check that in the loop condition and handle it somehow if we get an EOF, but the size is still below the actual file_size?